draft-ietf-jsonpath-base.md

```diff
@@ -1361,9 +1362,47 @@ Provisional registration? (standards tree only):
 
 # Security Considerations {#Security}
 
-This section gives security considerations, as required by {{RFC3552}}.
-
-
+Security considerations for JSONPath can stem from
+
+* attack vectors on JSONPath implementations, and
+* the way JSONPath is used in security-relevant mechanisms.
+
+## Attack vectors on JSONPath Implementations
+
+Historically, JSONPath has often been implemented by feeding parts of
+the query to an underlying programming language engine, e.g.,
+JavaScript.
+This approach is well known to lead to injection attacks and would
+require perfect input validation to prevent these attacks (see
+{{Section 12 of -json}} for similar considerations for JSON itself).
+Instead, JSONPath implementations need to implement the entire syntax
+of the query without relying on the parsers of programming language
+engines.
+
+Attacks on availability may attempt to trigger unusually expensive
+runtime performance exhibited by certain implementations in certain
+cases.
+(See {{Section 10 of -cbor}} for issues in hash-table implementations,
+and {{Section 8 of -iregexp}} for performance issues in regular
+expression implementations.)
+Implementers need to be aware that good average performance is not
+sufficient as long as an attacker can choose to submit specially
+crafted JSONPath queries that trigger surprisingly high, possibly
+exponential, CPU usage.
+
+## Attacks on Security Mechanisms that Employ JSONPath
+
+Where JSONPath is used as a part of a security mechanism, attackers
+can attempt to evoke unexpected behavior, or take advantage of
+differences in behavior between implementations that implement some
+protection and the implementations that are being protected by these.
+
+This also applies to underlying technologies such as UTF-8 (see
+{{Section 10 of -utf8}}), the Unicode character set, and JSON.
+A characteristic of JSON that can lead to diverging results is the
```

Comment:
"diverging" carries connotations of failure. Suggestion:
```suggestion
A characteristic of JSON that can lead to varying results is the
```

Reply:
Another option:
```suggestion
A characteristic of JSON that can lead to non-deterministic results is the
```

Reply:
Well, it *is* failure :-)  But varying is fine, too.  Didn't want to spend "non-deterministic" here, because for some people that sounds like someone is throwing a coin (which, however, can figuratively exactly what's happening).